### ISSUE TITLE
Can't mark message as seen.

### ISSUE THREAD

**Issue by cmichal** [mscdex/node-imap#430]
I'm trying to mark message as seen via fetch method 'markSeen:true' attribut or via setFlags/addFlags but when I fetch them for the second time they are still NEW.

If I use markSeen:true and display body of the message I can see the flag is there, but it's not persistent.

My message box is not read only and I can delete message by adding appropriate flag and closing the mailbox. Also flag '\Seen' is on box permFlags list. I am using version 0.8.14.

thanks
mC

**Comment by mkozjak** [comment]
Same here.

**Comment by polonel** [comment]
+1

**Comment by benqiang** [comment]
+1

**Comment by alamcordeiro** [comment]
+ 1

**Comment by thiagopio** [comment]
+1

**Comment by armw4** [comment]
+1

**Comment by Mornor** [comment]
+1

**Comment by pennymax** [comment]
+1...

**Comment by somename85** [comment]
+1

**Comment by Fruity84** [comment]
You should call addFlags with the **UID** and not the **Seqno**.
If you want mark them as 'seen' by seqno call imap.**seq**.fetch([seqno], function() {
markSeen: true
}

**Comment by vflyson** [comment]
Here's how node-imap does it https://github.com/mscdex/node-imap/blob/master/lib/Connection.js#L835 during .fetch(). And Gmail/whatever doesn't seem to support the .PEEK flag for fetching emails. So that makeSeen option is useless.

Solution: https://github.com/xc-stars/xcmail/blob/cb797ebe877fb3c7309b7bc4d248f7ecaec2f13b/src/main-process/mailReceiver.js#L118

You basically will need to be manually setting that \SEEN flag.
I do it like this:
```
msg.once('attributes', function(attrs) {
    markMailSeen(attrs.uid)
});
```

Make a note that Seqno is not UID (both are sequiential digits, yet you need the UID because that's the unique one).

**Comment by anks333** [comment]
`imap.openBox("INBOX", false, function(err, mailBox) {
        if (err) {
            console.error(err);
            return;
        }
        imap.search(["UNSEEN"], function(err, results) {

            imap.setFlags(results, ['\\Seen'], function(err) {
                if (!err) {
                    console.log("marked as read");
                } else {
                    console.log(JSON.stringify(err, null, 2));
                }
            });

            var f = imap.seq.fetch(results, { bodies: "" });
            f.on("message", processMessage);
            f.once("error", function(err) {
                return Promise.reject(err);
            });
            f.once("end", function() {
                console.log("Done fetching all unseen messages.");
                imap.end();
            });
        });
    });`

processMessage

`function processMessage(msg seqno){
         //use mailparser
}`


I don't know why my code is not getting properly formatted here,
so basically results contains the uuid

we need to call imap.setFlags([uuids], ['\\Seen'], cb) function to make it mark as read


handle all errors properly, that i haven't done here :
